edgelet/iotedged/src/lib.rs

```diff
@@ -669,7 +669,9 @@ where
     .context(ErrorKind::Initialize(
         InitializeErrorReason::DpsProvisioningClient,
     ))?;
-    let provision = dps
+    let provision_with_file_backup = BackupProvisioning::new(dps, backup_path);
```

Comment:
Tests? To make sure the backup is created/used/overwritten, etc?

Reply:
I will have to refactor the code a bit more to get the tests enabled. Currently the code structure does not lend itself to testing very well. I will make these updates in a separate PR. For now manual E2E verification of both TPM and Symmetric key auth has been done.